src/ext-test/java/org/opentripplanner/ext/emissions/EmissionsTest.java

```diff
@@ -170,4 +170,91 @@ void testZeroEmissionsForItineraryWithZeroEmissions() {
       emissionsFilter.filter(List.of(i)).get(0).getEmissionsPerPerson().getCo2()
     );
   }
+
+  @Test
+  void testGetEmissionsForCombinedRoute() {
+    var stopOne = TransitModelForTest.stopForTest("1:stop1", 60, 25);
+    var stopTwo = TransitModelForTest.stopForTest("1:stop1", 61, 25);
+    var stopThree = TransitModelForTest.stopForTest("1:stop1", 62, 25);
+    var stopPattern = TransitModelForTest.stopPattern(stopOne, stopTwo, stopThree);
+    var route = TransitModelForTest.route(id("1")).build();
+    var pattern = TransitModelForTest.tripPattern("1", route).withStopPattern(stopPattern).build();
+    var stoptime = new StopTime();
+    var stoptimes = new ArrayList<StopTime>();
+    stoptimes.add(stoptime);
+    var trip = Trip
+      .of(FeedScopedId.parse("FOO:BAR"))
+      .withMode(TransitMode.BUS)
+      .withRoute(route)
+      .build();
+    var transitLeg = new ScheduledTransitLegBuilder<>()
```

Comment:
Maybe add some utility method in this class for creating transitLeg/carLeg as it is done multiple times now.